tests/development-setup.yml

```diff
@@ -0,0 +1,137 @@
+    - name: Include Libreswan vars
+      include_vars: "../playbooks/roles/l2tp-ipsec/vars/main.yml"
+
+    - name: Install the Libreswan dependencies that are required for compilation
+      apt:
+        name: "{{ item }}"
+      with_items: "{{ libreswan_compilation_dependencies_trusty }}"
+      when: ansible_distribution == 'Ubuntu' and ansible_distribution_release == 'xenial'
+
+    - name: Install xmlto (without the extraordinarily large number of packages it normally recommends)
+      apt:
+        name: xmlto
+        install_recommends: no
+
+    - name: Retrieve the Libreswan source code
+      get_url:
+        url: https://d25kfp60e9u1dw.cloudfront.net/{{ libreswan_source_filename }}
+        dest: "{{ libreswan_source_location }}"
+        checksum: "{{ libreswan_checksum }}"
+        owner: root
+        group: root
+        mode: 0644
+      register: libreswan_get_result
+      until: libreswan_get_result|success
+      retries: 3
+      delay: 5
+
+    - name: Extract the Libreswan source code
+      unarchive:
+        copy: no
+        src: "{{ libreswan_source_location }}"
+        dest: /usr/local/src
+        owner: root
+        group: root
+        creates: "{{ libreswan_compilation_directory }}/README"
+
+    - name: Compile and install Libreswan (this will take a while)
+      shell: make -j {{ ansible_processor_cores }} programs && make install
+      args:
+        chdir: "{{ libreswan_compilation_directory }}"
+
+    - name: Start ipsec
+      service:
+        name: ipsec
+        state: started
+        enabled: yes
+      register: ipsec_start
+
+    - name: Pause for a few seconds to let ipsec start
+      pause:
+        seconds: 5
+
+    - name: Install lxd
+      apt:
+        name: lxd
+        state: present
+        update_cache: yes
+
+    - name: lxd new group
+      shell: newgrp lxd
+
+    - block:
+        - name: lxd init config
+          shell: lxd init --auto --storage-backend dir
+
+        - name: lxd create network
+          shell: lxc network create testbr0
+
+        - name: lxd attach network to default profile
+          shell: lxc network attach-profile testbr0 default eth0
+      ignore_errors: yes
+
+    - name: restart lxd
+      service:
+        name: lxd
+        state: restarted
+
+    - name: pause
+      pause:
+        seconds: 10
+
+    - name: set permission on lxd daemon
+      file:
+        path: /var/lib/lxd/unix.socket
+        mode: "0777"
+
+    - name: Launch streisand container
+      lxd_container:
+        name: streisand
+        state: started
+        source:
+          type: image
+          mode: pull
+          server: https://images.linuxcontainers.org
+          protocol: lxd
+          alias: ubuntu/xenial/amd64
+        profiles: ["default"]
+        config:
+          security.privileged: "true"
+        wait_for_ipv4_addresses: true
+        timeout: 300
```

Comment:
It might be worth adding a note to the task name here that this can take a long time. It was a pretty considerable wait (>60s? ) for me on initial run.